protocol/src/reqres.rs

```diff
@@ -48,61 +53,50 @@ impl RequestResponseCodec for ReqRespCodec {
     type Request = Req;
     type Response = Resp;
 
-    async fn read_request<T>(
-        &mut self,
-        _: &ReqRespProtocol,
-        io: &mut T,
-    ) -> io::Result<Self::Request>
+    async fn read_request<T>(&mut self, _: &ReqRespProtocol, io: &mut T)
+        -> io::Result<Self::Request>
     where
-        T: AsyncRead + Unpin + Send,
+        T: AsyncRead + Unpin + Send
     {
         read_one(io, 1024)
             .map(|res| match res {
```

Comment:
Probably we should check the current session state to be sure that get a response for the corresponding request. For example, right now we could get `Pong` for `Get` request without any errors, that's wrong.